src/components/EventCard.tsx

```diff
@@ -11,12 +11,26 @@ import { getEvents_getEvents } from '../generated/getEvents';
 
 interface Props {
   event: getEvents_getEvents;
+  addModalContent: (
+    title: string,
+    location: string,
+    date: string,
+    nonprofit: any,
+    id: number,
+  ) => void;
+  toggleModal: () => void;
 }
 
-const EventCard = ({ event }: Props) => {
-  const { title, location, date, nonprofit, image } = event;
+const EventCard = ({ event, addModalContent, toggleModal }: Props) => {
+  const updateModal = () => {
```

Comment:
need to define this after you destructure them out, suprised TS didn't fail here.  That said, this approach doesn't seem great.  these are all props that are being passed into this card, so it shouldn't need to pass them back into the event handler, you can do that in the component that uses EventCard